modules/source-bookshelf/lib/process_sort.js

```diff
@@ -3,11 +3,12 @@ const _ = require('lodash');
 // TODO: investigate how to deal with express's query parser
 // converting + into a space.
 function isAscending (key) {
-  return key[0] === '+' || key[0] === ' ';
+  return key[0] === '+' || key[0] === '-';
 }
 
 module.exports = function (model, query, sortBy) {
   var validFields = model.fields;
+  console.log(model.fields);
```

Comment:
They are not required, this is part of the many hacks I've done to Bookshelf to make endpoints even be possible. It is possible for bookshelf to introspect this, but it can't be done sync, and it can't be done at extension time, so every interaction with the models needs to be behind an async flag. Long story short, eventually the fields listing won't be required, but presently it is.